test/unit/voldemort/client/OfflineStateTest.java

```diff
@@ -262,4 +268,23 @@ public void testStateTransitions() throws InterruptedException {
         assertTrue(testOnlineTraffic());
         assertTrue(testSlopStreaming());
     }
+
+    @Test
+    public void testStartingOfflineServeres() throws Exception {
+        VoldemortServer server = getVoldemortServer(0);
+        server.goOffline();
+        server.stop();
+
+        String baseDirPath = server.getVoldemortConfig().getVoldemortHome();
+        String parentDirPath = Paths.get(baseDirPath).getParent().toString();
+        List<StoreDefinition> storeDefs = server.getMetadataStore().getStoreDefList();
+        try {
+            VoldemortConfig config =
+                    ServerTestUtils.createServerConfigWithDefs(useNio, 0, parentDirPath, cluster, storeDefs, new Properties());
+            VoldemortServer newServer = ServerTestUtils.startVoldemortServer(socketStoreFactory, config, cluster);
+            fail("should have thrown an connection exception");
+        }catch (VoldemortException e){
```

Comment:
I would just handle Catch UnreachableStoreException and also check that the inner exception is ConnectException.class.